include/alpaka/core/CallbackThread.hpp

```diff
@@ -61,8 +61,9 @@ namespace alpaka::core
                 m_tasks.emplace(std::move(task));
                 if(!m_thread.joinable())
                     startWorkerThread();
+                m_cond.notify_one();
             }
-            m_cond.notify_one();
+
```

Comment:
Is that a necessary change? AFAIK notifying a condition variable outside the lock is actually the preferred implementation, since the notified thread can immediately acquire the lock.

Reply:
Yes it is the same as in the de-structor we can miss the notification. The sanitizer issues got solved by this line.

Reply:
Ah, I remember.  Alright then!